test-suite/prices.cpp

```diff
@@ -0,0 +1,218 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ Copyright (C) 2004 StatPro Italia srl
+
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+#include "toplevelfixture.hpp"
+#include "utilities.hpp"
+#include <ql/prices.hpp>
+#include <ql/time/date.hpp>
+#include <array>
+
+using namespace QuantLib;
+using namespace boost::unit_test_framework;
+
+BOOST_FIXTURE_TEST_SUITE(QuantLibTest, TopLevelFixture)
+
+BOOST_AUTO_TEST_SUITE(PricesTest)
+
+namespace {
+    namespace tt = boost::test_tools;
+
+    BOOST_AUTO_TEST_CASE(testMidEquivalent) {
+        BOOST_TEST_MESSAGE("Testing midEquivalent()...");
+
+        BOOST_TEST(1.5 == midEquivalent(1, 2, 3, 4), tt::tolerance(1e-14));
+        BOOST_TEST(1.5 == midEquivalent(1, 2, 0, 4), tt::tolerance(1e-14));
+        BOOST_TEST(1.5 == midEquivalent(1, 2, 3, 0), tt::tolerance(1e-14));
+        BOOST_TEST(1.5 == midEquivalent(1, 2, 0, 4), tt::tolerance(1e-14));
```

Comment:
This is the same test as the second line.  Was it meant to be something else?  The same goes for the groups below.

Reply:
Fixed in f8cbb3690b73da0bd15e3019129aa5b3a8c5e4e4.